### PR TITLE
Fix/text containing finder work with rich texts

### PR DESCRIPTION
**Pull request by ghost**
`textContaining` finder does not go through `RichText`.

A fix on the `text` finder has been done but the changes did not impact `textContaining` finder.

So I've made 2 changes, first I've made the changes on `textContaining` finder in order to make it go through `RichText`
And in a second commit I've linked `text` and `textContaining` finders to be sure the future changes will be done on both of these finders.

This is linked to this issue :
https://github.com/flutter/flutter/issues/99681

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by goderbauer** [comment]
Thanks for the changes. Looks like the analyzer check is unhappy know. Can You take a look?

File: packages/flutter_test/lib/src/finders.dart [hunk 1]
```diff
@@ -77,7 +77,17 @@ class CommonFinders {
   ///
   /// If the `skipOffstage` argument is true (the default), then this skips
   /// nodes that are [Offstage] or that are from inactive [Route]s.
-  Finder textContaining(Pattern pattern, { bool skipOffstage = true }) => _TextContainingFinder(pattern, skipOffstage: skipOffstage);
+  Finder textContaining(
+    Pattern pattern, {
+    bool findRichText = false,
```
[the remark this answers is not in the document]
Comment (valentinmarq) [reply packages/flutter_test/lib/src/finders.dart:102]: Thank you for your feedbacks !

I've updated it

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by valentinmarq** [comment]
Hi @dnfield, can you take a look at my PR ?
(I don't know if it's ok to ask for a review with a comment but I can't request a review from anyone through the usual menu)

**Review by dnfield — APPROVED** [review]

**Pull request merged**